### ISSUE TITLE
Failure to install the mandeR 

### ISSUE THREAD

**Issue by heatheranndye** [gerrymandr/mandeR#1]
install_github('r-barnes/mandeR', vignette=TRUE)
Downloading GitHub repo r-barnes/mandeR@master
from URL https://api.github.com/repos/r-barnes/mandeR/zipball/master
Installing mandeR
Installing 1 package: Rcpp
Installing package into ‘C:/Users/Heather/Documents/R/win-library/3.4’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/Rcpp_0.12.13.zip'
Content type 'application/zip' length 4354267 bytes (4.2 MB)
downloaded 4.2 MB

package ‘Rcpp’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Heather\AppData\Local\Temp\RtmpkXuw3n\downloaded_packages
Installing 1 package: sf
Installing package into ‘C:/Users/Heather/Documents/R/win-library/3.4’
(as ‘lib’ is unspecified)
also installing the dependencies ‘udunits2’, ‘DBI’, ‘units’, ‘magrittr’

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/udunits2_0.13.zip'
Content type 'application/zip' length 277648 bytes (271 KB)
downloaded 271 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/DBI_0.7.zip'
Content type 'application/zip' length 748921 bytes (731 KB)
downloaded 731 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/units_0.4-6.zip'
Content type 'application/zip' length 806752 bytes (787 KB)
downloaded 787 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/magrittr_1.5.zip'
Content type 'application/zip' length 155741 bytes (152 KB)
downloaded 152 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.4/sf_0.5-4.zip'
Content type 'application/zip' length 33154989 bytes (31.6 MB)
downloaded 31.6 MB

package ‘udunits2’ successfully unpacked and MD5 sums checked
package ‘DBI’ successfully unpacked and MD5 sums checked
package ‘units’ successfully unpacked and MD5 sums checked
package ‘magrittr’ successfully unpacked and MD5 sums checked
package ‘sf’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Heather\AppData\Local\Temp\RtmpkXuw3n\downloaded_packages
"C:/PROGRA~1/R/R-34~1.2/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/Heather/AppData/Local/Temp/RtmpkXuw3n/devtools241856796cd3/r-barnes-mandeR-a0c7496"  \
  --library="C:/Users/Heather/Documents/R/win-library/3.4" --install-tests 

* installing *source* package 'mandeR' ...
** libs

*** arch - i386
C:/RBuildTools/3.4/mingw_32/bin/g++  -std=gnu++11 -c     -O2 -Wall  -mtune=core2 -I"C:/PROGRA~1/R/R-34~1.2/include" -DNDEBUG -DDOCTEST_CONFIG_DISABLE -I"C:/Users/Heather/Documents/R/win-library/3.4/Rcpp/include"   -I"d:/Compiler/gcc-4.9.3/local330/include" -DDOCTEST_CONFIG_DISABLE Rwrapper.cpp -o Rwrapper.o 
Rwrapper.cpp:2:45: fatal error: compactnesslib/compactnesslib.hpp: No such file or directory
 #include "compactnesslib/compactnesslib.hpp"
                                             ^
compilation terminated.
make: *** [Rwrapper.o] Error 1
Warning: running command 'make -f "Makevars.win" -f "C:/PROGRA~1/R/R-34~1.2/etc/i386/Makeconf" -f "C:/PROGRA~1/R/R-34~1.2/share/make/winshlib.mk" CXX='$(CXX11) $(CXX11STD)' CXXFLAGS='$(CXX11FLAGS)' CXXPICFLAGS='$(CXX11PICFLAGS)' SHLIB_LDFLAGS='$(SHLIB_CXX11LDFLAGS)' SHLIB_LD='$(SHLIB_CXX11LD)' SHLIB="mandeR.dll" ' had status 2
ERROR: compilation failed for package 'mandeR'
* removing 'C:/Users/Heather/Documents/R/win-library/3.4/mandeR'
Installation failed: Command failed (1)
Warning message:
GitHub repo contains submodules, may not function as expected!

**Comment by sastoudt** [comment]
I had this same issue, and @mahrud helped me fix it.

In terminal: 
``` 
git clone https://github.com/gerrymandr/mandeR.git
cd mandeR

git submodule init
git submodule update

cd ..
tar czf mandeR.tar.gz mandeR
```

Then in R:
```
wd="" ## change to where you have mandeR.tar.gz
setwd(wd)

install.packages("mander.tar.gz", repos = NULL, type = "source")
```

**Comment by migurski** [comment]
Fixed in #2.

**Comment by r-barnes** [comment]
This was probably due to an out-of-date compactnesslib.

I've now modified things to take the newest version. Unfortunately, this has introduced a Boost dependency. I haven't tried installing on other machines, yet.
